### PR TITLE
Asv test support

### PR DESCRIPTION
**Pull request by jperez999**
These changes add support for benchmarking leveraging the Air Speed Velocity framework to display benchmarks over time. This will be extremely helpful in catch regressions in performance while also ensuring any new changes do not adversely affect our selected benchmarks.

### CODE REVIEW

**Comment by jperez999** [comment]
rerun tests

**Comment by jperez999** [comment]
rerun tests

File: examples/rossmann/rossmann-store-sales-fastai.ipynb [hunk 1]
```diff
@@ -296,6 +297,7 @@
     "from fastai.basics import Learner\n",
     "from fastai.basics import MSELossFlat\n",
     "from fastai.callback.progress import ProgressCallback\n",
+    "from fastai.metrics import APScoreMulti, RocAucMulti\n",
```
Comment (benfred) [inline examples/rossmann/rossmann-store-sales-fastai.ipynb:300]: Are these imports used anywhere? I don't see it reference in the rest of this PR

File: tests/conftest.py [hunk 2]
```diff
@@ -173,6 +178,43 @@ def dataset(request, paths, engine):
     return nvtabular.Dataset(paths, part_mem_fraction=gpu_memory_frac, **kwargs)
 
 
+@pytest.fixture(scope="session")
+def db():
```
Comment (benfred) [inline tests/conftest.py:182]: As a test fixture name - 'db' is pretty general. Maybe we should rename to 'asv_db' or something that indicates this is specific to asv?

File: tests/integration/benchmark_parsers.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    def get_dl_thru(self, full_time, num_rows, epochs, throughput):
+        return create_bench_result(
+            f"{self.name}_dataloader",
+            [("time", full_time), ("rows", num_rows), ("epochs", epochs)],
+            throughput,
+            "rows/second",
+        )
+
+    def bres_loss(self, epoch, loss, l_type="train"):
```
Comment (benfred) [inline tests/integration/benchmark_parsers.py:25]: nitpick:  I don't think 'bres' adds anything to the name - could we just rename to 'loss' ? If you want to make it clear thats its a BenchmarkResult being returned , we could add type annotations here =) 
```suggestion
    def loss(self, epoch, loss, l_type="train") -> BenchmarkResult:
```

File: tests/integration/test_notebooks.py [hunk 4]
```diff
@@ -43,57 +46,69 @@ def test_criteo_notebook(tmpdir):
...
-        transform=lambda line: line.replace("rmm.reinitialize(", "# rmm.reinitialize("),
-        gpu_id="0,1",
-        batch_size=100000,
-    )
-
-
-def test_optimize_criteo(tmpdir):
-    input_path = os.path.join(DATA_START, "criteo/crit_orig")
-    output_path = os.path.join(DATA_START, "criteo/crit_test_opt")
+# def test_criteohugectr_notebook(tmpdir):
```
Comment (benfred) [inline tests/integration/test_notebooks.py:54]: Did you mean to comment out this test?

File: tests/unit/test_notebooks.py [hunk 5]
```diff
@@ -61,8 +61,15 @@ def test_rossman_example(tmpdir):
...
+    notebook_path = os.path.join(
+        dirname(TEST_PATH), "examples/rossmann/", "rossmann-store-sales-feature-engineering.ipynb"
+    )
+    _run_notebook(tmpdir, notebook_path)
+
+    os.environ["INPUT_DATA_DIR"] = str(tmpdir)
+
     notebooks = [
-        "rossmann-store-sales-feature-engineering.ipynb",
+        #         "rossmann-store-sales-feature-engineering.ipynb",
```
Comment (benfred) [inline tests/unit/test_notebooks.py:72]: Can we just delete this line ?

File: tests/integration/test_notebooks.py (hunk 4 above)
Comment (jperez999) [reply tests/integration/test_notebooks.py:54]: Yes the hugectr test is commented out. Wanted to integrate it in a seperate PR, but I didnt want to loose the code.

**Review by benfred — APPROVED** [review]

**Comment by jperez999** [comment]
rerun tests

**Pull request merged**